tests/test_helper.hpp

```diff
@@ -140,3 +140,16 @@ bool check_results(sycl::ext::cplx::complex<T> output,
   }
   return true;
 }
+
+template <typename T>
+bool check_results(T output, T reference, bool is_device) {
+  if (!almost_equal_scalar(output, reference, SYCL_CPLX_TOL_ULP)) {
+    std::cerr << std::setprecision(std::numeric_limits<T>::max_digits10)
+              << "Test failed with complex_type: " << get_typename<T>()
+              << " Computed on " << (is_device ? "device" : "host")
+              << " Output: " << output << " Reference: " << reference
+              << std::endl;
+    return false;
+  }
+  return true;
+}
```

Comment:
Should we merge `almost_equal_scalar`and `almost_equal_cplx`  into one general `almost_equal`?
Then we can remove this function I think

Reply:
There were some issues that caused me to separate this. I believe the issue is that the floating point types, can be implicitly converted into a complex type. Therefore it makes it ambiguous as to if the compiler should the complex or floating point type.

Reply:
Oh, then perfect nothing to change!
I will do some tests on my side, but let merge this PR :) 

Thanks again!